From 0eb437a11aa5efa7fa545f2796fc65e34f4fc83a Mon Sep 17 00:00:00 2001
From: Bin Han <bin.b.han@intel.com>
Date: Wed, 3 Dec 2014 14:34:15 +0800
Subject: [PATCH 175/176] atomisp2: skip a FW event without valid pipe

BZ: 230100

There is some new FW event which is not with a valid pipeline, in
this case, driver should just skip it(not handle it) instead of
failing on it.

Change-Id: I37a7951f564626a13f9bddddc0cfc0cc2e115872
Tracked-On:https://jira01.devtools.intel.com/browse/IMINAN-12259
Signed-off-by: Bin Han <bin.b.han@intel.com>
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/32170
Integrator: Wei Tao <wei.tao@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index f9072f3..ad51856 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -4575,9 +4575,9 @@ int atomisp_css_isr_thread(struct atomisp_device *isp,
 		asd = __get_atomisp_subdev(current_event.event.pipe,
 					isp, &stream_id);
 		if (!asd) {
-			dev_err(isp->dev, "%s:no subdev.event:%d",  __func__,
-					current_event.event.type);
-			return -EINVAL;
+			dev_warn(isp->dev, "%s:no subdev.event:%d",  __func__,
+				current_event.event.type);
+			continue;
 		}
 
 		atomisp_css_temp_pipe_to_pipe_id(asd, &current_event);
@@ -4636,6 +4636,12 @@ int atomisp_css_isr_thread(struct atomisp_device *isp,
 			break;
 		}
 	}
+	/* There is some new FW event which is not with a valid pipeline,
+	 * in this case, driver should just skip it(not handle it) instead of
+	 * failing on it.
+	 */
+	if (!asd)
+		return 0;
 	/* If there are no buffers queued then
 	 * delete wdt timer. */
 	if (!atomisp_buffers_queued(asd))
-- 
1.9.1

